airbyte-cdk/python/airbyte_cdk/test/mock_http/request.py

```diff
@@ -41,15 +44,16 @@ def matches(self, other: Any) -> bool:
                 and self._parsed_url.hostname == other._parsed_url.hostname
                 and self._parsed_url.path == other._parsed_url.path
                 and (
-                    ANY_QUERY_PARAMS in [self._query_params, other._query_params]
+                    ANY_QUERY_PARAMS in (self._query_params, other._query_params)
                     or parse_qs(self._parsed_url.query) == parse_qs(other._parsed_url.query)
                 )
                 and _is_subdict(other._headers, self._headers)
+                and other._body == self._body
```

Comment:
I'm a bit confused as to how we want to approach this. I think the current approach is lacking as illustrated by the following test:
```
    @HttpMocker()
    def test_given_body_is_json_when_decorate_then_match_query(self, http_mocker):
        http_mocker.get(
            HttpRequest(_A_URL, _SOME_QUERY_PARAMS, _SOME_HEADERS, {"a body": "body"}),
            _A_RESPONSE,
        )

        requests.get(_A_URL, params=_SOME_QUERY_PARAMS, headers=_SOME_HEADERS, json={"a body": "body"})

        # should fail if not match
```

This test current fails. However, the user will have set up the same mapping in the mock than in the tested code (in this case, the tested code is `requests.get(...)`). I'm wondering if there are other cases we should be wary of. Do you mind digging deeper into that and propose a list of cases we would support? I would like this to be documented somewhere if possible (either as tests or as a class comment)

Reply:
I think the current solution is better. Should we also handle cases like this?

```
    @HttpMocker()
    def test_given_body_is_json_when_decorate_then_match_query(self, http_mocker):
        http_mocker.get(
            HttpRequest(_A_URL, _SOME_QUERY_PARAMS, _SOME_HEADERS, {"a body": "body"}),
            _A_RESPONSE,
        )

        requests.get(_A_URL, params=_SOME_QUERY_PARAMS, headers=_SOME_HEADERS, data='{\n  "a body": "body"\n}')

        # should fail if not match
```

Technically, they are the same and it's just how the json object was formatted that differs. If the formatting would be important for whatever reason, I would expect the dev to initialize `HttpRequest` with a string and not an object

Reply:
Added support for this case. Please take a look